### PR TITLE
feature/authentication

### PR DESCRIPTION
**Pull request by reetesh2001**
authentication

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,14 @@
 class ApplicationController < ActionController::Base
+    before_action :authenticate_user!
```
Comment (Shubham-Porwall) [inline app/controllers/application_controller.rb]: Please fix indentation we use 2 space indentation

File: app/controllers/hrs_controller.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+class HrsController < ApplicationController
+    skip_before_action :authenticate_user!, only: [:new, :create]
+
+    def show
+        @hr = Hr.find(params[:id])
```
Comment (Shubham-Porwall) [inline app/controllers/hrs_controller.rb]: Why Find?

File: app/controllers/hrs_controller.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+class HrsController < ApplicationController
+    skip_before_action :authenticate_user!, only: [:new, :create]
```
Comment (Shubham-Porwall) [inline app/controllers/hrs_controller.rb]: Why you skip autho before creating ? Please specify reason

File: app/views/hrs/show.html.erb [hunk 4]
```diff
@@ -0,0 +1 @@
+<h2>hr path show</h2>
```
Comment (Shubham-Porwall) [inline app/views/hrs/show.html.erb]: Please leave one line empty at End of the file

File: db/migrate/20240718063323_create_users.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+class CreateUsers < ActiveRecord::Migration[7.1]
+  def change
+    create_table :users do |t|
```
Comment (Shubham-Porwall) [inline db/migrate/20240718063323_create_users.rb:3]: What is the use of user table ?

File: test/controllers/hrs_controller_test.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (Shubham-Porwall) [inline test/controllers/hrs_controller_test.rb]: Why this file added with this commit have you worked on this ?

File: app/controllers/hrs_controller.rb (hunk 2 above)
Comment (reetesh2001) [reply app/controllers/hrs_controller.rb]: sorry for that,I have fixed by using find_by now.

File: app/controllers/hrs_controller.rb (hunk 3 above)
Comment (reetesh2001) [reply app/controllers/hrs_controller.rb]: if I do not skip before creating, data will not be saved.

File: db/migrate/20240718063323_create_users.rb (hunk 5 above)
Comment (reetesh2001) [reply db/migrate/20240718063323_create_users.rb:3]: there is no need of user it happend by mistake, when i was creating hr model.

File: test/controllers/hrs_controller_test.rb (hunk 6 above)
Comment (reetesh2001) [reply test/controllers/hrs_controller_test.rb]: No,I did not work.

File: db/migrate/20240718095751_drop_user_table.rb [hunk 7]
```diff
@@ -0,0 +1,14 @@
+class DropUserTable < ActiveRecord::Migration[7.1]
```
Comment (sdgupta94) [inline db/migrate/20240718095751_drop_user_table.rb:1]: question: why we have created 2 migration for user deletion?

Comment (reetesh2001) [reply db/migrate/20240718095751_drop_user_table.rb:1]: in the first migration, there was a mistake by me, so it did not delete user table ,the mistake was ,i put down methed change table

**Pull request merged**